src/sentry_transport.c

```diff
@@ -5,9 +5,11 @@
 #include "sentry_ratelimiter.h"
 #include "sentry_string.h"
 
+#include "external/crashpad/third_party/zlib/zlib/zlib.h"
```

Comment:
Yeah, @Strive-Sun, as you mentioned, that is problematic. The transport implementation is entirely independent of any backend being built, so a direct dependency on a particular backend doesn't work.

If you want to continue this PR, I would recommend to:

* search for an installed `zlib` in the top-level `CMakeLists.txt` (but do not make it a required dependency)
* if `zlib` was found, use a compile definition that enables your compression code in the transport and uses the system-include instead of the `crashpad` one
* make sure that the `crashpad` build uses the same `zlib` implementation (because it will be linked with `libsentry` via `crashpad-util`) by setting the `CRASHPAD_ZLIB_SYSTEM` option in the top-level `CMakeLists.txt` before `add_subdirectory()` for `crashpad`.

This way, the transport compression will be enabled whenever a `zlib` system implementation is available, and we have no clash in case `crashpad` is used as a backend.

An alternative path to the first point in my recommendations would be to introduce a build-option `SENTRY_TRANSPORT_COMPRESSION`, where you can make the dependency required after checking for that option. The remaining points would be the same. The difference would be that people can opt-in/-out from compression and don't get that feature simply by an existing dependency on their build-system.

Reply:
@supervacuus Thank you for your detailed suggestions. I added the option `SENTRY_TRANSPORT_COMPRESSION` to `CMakeLists.txt`, which make the dependency required after checking for that option. But now there are two questions.

- whether `zlib` needs to add the `cmake < 3.12` support like the libcurl?
- The default installation method for third-party libraries in Windows is to use the `vcpkg` tool, which is a universal approach. Dependency cannot find the library installed by `vcpkg`, how can find these libs?
![image](https://github.com/getsentry/sentry-native/assets/50010647/ea6b5b20-ec9d-4f92-a18c-84ebb6969beb)

Reply:
> whether zlib needs to add the cmake < 3.12 support like the libcurl?

No. We no longer support CMake 3.12 (except on Android, where no transport is used). Let's keep it slim.

> The default installation method for third-party libraries in Windows is to use the vcpkg tool, which is a universal approach. Dependency cannot find the library installed by vcpkg, how can find these libs?

`SENTRY_TRANSPORT=curl` on Windows is not supported. Please also consider that the dependency was found, but the `AsynchDNS` feature wasn't.

Reply:
@supervacuus I updated this pr, add `zlib.h` in `sentry_transport.c` and add `SENTRY_TRANSPORT_COMPRESSION`  in CMakeLists. It should be noted that, for compilation under Windows, setting `CMAKE_TOOLCHAIN_FILE` is required if the `zlib` library installed via vcpkg is needed.